service/controller/key/machine_deployment.go

```diff
@@ -1,12 +1,239 @@
 package key
 
 import (
+	"fmt"
 	"strconv"
 
 	infrastructurev1alpha2 "github.com/giantswarm/apiextensions/pkg/apis/infrastructure/v1alpha2"
 	"github.com/giantswarm/microerror"
 
 	"github.com/giantswarm/aws-operator/pkg/annotation"
+	"github.com/giantswarm/aws-operator/service/controller/resource/tcnp/template"
+)
+
+var (
+	MachineDeploymentLaunchTemplateOverrides = map[string][]template.LaunchTemplateOverride{
```

Comment:
@MarcelMue @anvddriesch where does this come from? Do we ever have to adjust this? I would like to add a nice little comment.

Reply:
Yes - this is currently the source of truth for "equal machines" when taking spot instances. So when AWS fills up our spot instances with `m4.xlarge` (line 16) then we give it the option to either fill with `m4.xlarge` or `m5.xlarge`. That way AWS can take the cheaper one of the two for the moment.

So in the future this should either be moved into some more sensible location (e.g. an `AWSInfoService`) or simply remain here. It is likely to change with added overrides etc.

@marians should also know more from the UX side.

Reply:
So that means we made up this mapping and can adapt it to our needs on a per use case basis.

Reply:
Yes - it was spawned out of our mind during the hackathon.